pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -313,13 +315,30 @@ def store_predictions(self, output, batch_idx, dataloader_idx):
 
     def on_evaluation_epoch_end(self, *args, **kwargs):
         # call the callback hook
-        if self.trainer.testing:
-            self.trainer.call_hook('on_test_epoch_end', *args, **kwargs)
-        else:
-            self.trainer.call_hook('on_validation_epoch_end', *args, **kwargs)
+        self.call_on_evaluation_epoch_end_hook()
 
         self.trainer.call_hook('on_epoch_end')
 
+    def call_on_evaluation_epoch_end_hook(self):
```

Comment:
this seems to be not tested... (at least codecov complains about it)

Reply:
Yeah, let me look into it.

Reply:
Updated!